packages/replay/src/session/Session.ts

```diff
@@ -24,59 +23,48 @@ interface SessionObject {
   segmentId: number;
 
   /**
-   * Is the session sampled? `null` if the sampled, otherwise, `session` or `error`
+   * The ID of the previous session.
+   * If this is empty, there was no previous session.
    */
-  sampled: Sampled;
-}
-
-export class Session {
-  /**
-   * Session ID
-   */
-  public readonly id: string;
+  previousSessionId?: string;
 
   /**
-   * Start time of current session
+   * Is the session sampled? `false` if not sampled, otherwise, `session` or `error`
    */
-  public started: number;
-
-  /**
-   * Last known activity of the session
-   */
-  public lastActivity: number;
-
-  /**
-   * Sequence ID specific to replay updates
-   */
-  public segmentId: number;
+  sampled: Sampled;
+}
 
-  /**
-   * Previous session ID
-   */
-  public previousSessionId: string | undefined;
+/**
+ * Get a session with defaults & applied sampling.
+ */
+export function makeSession(session: Partial<Session> & { sampled: Sampled }): Session {
+  const now = new Date().getTime();
+  const id = session.id || uuid4();
+  const started = session.started || now;
+  const lastActivity = session.lastActivity || now;
+  const segmentId = session.segmentId || 0;
+  const sampled = session.sampled;
 
-  /**
-   * Is the Session sampled?
-   */
-  public readonly sampled: Sampled;
+  return {
+    id,
+    started,
+    lastActivity,
+    segmentId,
+    sampled,
+  };
+}
 
-  public constructor(session: Partial<SessionObject> = {}, { sessionSampleRate, errorSampleRate }: SampleRates) {
-    const now = new Date().getTime();
-    this.id = session.id || uuid4();
-    this.started = session.started ?? now;
-    this.lastActivity = session.lastActivity ?? now;
-    this.segmentId = session.segmentId ?? 0;
-    this.sampled =
-      session.sampled ?? (isSampled(sessionSampleRate) ? 'session' : isSampled(errorSampleRate) ? 'error' : false);
+/**
+ * Get the sampled status for a session based on sample rates & current sampled status.
+ */
+export function sampleSession(
+  sampled: Sampled | undefined,
```

Comment:
m: now with this refactor, isn't the only usage of `sampleSession` when it's undefined?

Does this mean we can even remove this arg and the early return?

Reply:
hmm, good point! Def. makes sense 👍